File: Year-2023/LeetCode-February-Challenge/ShuffletheArray.js
var shuffle = function(nums, n) {
    let res = [];
    for(let i = 0; i < n; ++i){
        res.push(nums[i]);
        if(i + n < nums.length){
            res.push(nums[i + n]);
        }
    }
    return res;
};
let nums = [2,5,1,3,4,7], n = 3;
console.log(shuffle(nums, n));

File: Year-2023/LeetCode-February-Challenge/ShuffletheArray.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> res;
        for(int i = 0; i < n; ++i){
            res.push_back(nums[i]);
            if(i + n < nums.size()){
                res.push_back(nums[i + n]);
            }
        }
        return res;
    }
};
int main(){
    vector<int> nums = {2,5,1,3,4,7};
    int n = 3;
    Solution *s = new Solution();
    vector<int> res = s->shuffle(nums, n);
    for(auto ele: res) cout<<ele<<" ";
    return 0;
}